zio/detector/file.go

```diff
@@ -79,7 +79,10 @@ func OpenParquet(zctx *resolver.Context, path string, opts zio.ReaderOpts) (*zbu
 
 func OpenFromNamedReadCloser(zctx *resolver.Context, rc io.ReadCloser, path string, opts zio.ReaderOpts) (*zbuf.File, error) {
 	var err error
-	r := GzipReader(rc)
+	r := io.Reader(rc)
+	if opts.Format != "zst" {
+		r = GzipReader(rc)
+	}
```

Comment:
Why special-case this?

Reply:
The zst reader uses type inference to pull the seeker interface out of the reader and wrapping the gzip reader hides the underlying interface.  It would be clearer to maybe have a method `zio.MustBeSeekable(opts.Format)` and wrap the gzip reader only when this is false.